pkg/git/git.go

```diff
@@ -126,17 +126,17 @@ func (c *Client) SourceBranch(commitHash string) (string, error) {
 // LatestTag returns the latest tag if found.
 func (c *Client) LatestTag() (string, error) {
 	result, err := c.Clean(c.Run("-C", c.repoDir, "tag", "--points-at", "HEAD", "--sort", "-version:creatordate"))
-	if err != nil {
-		return c.Clean(c.Run("-C", c.repoDir, "describe", "--tags", "--abbrev=0"))
+	if result != "" {
```

Comment:
Usually, there is no guarantee about the content of the first return parameter, in case the error is not nil. To be safe, I would do 2 checks here: 1. Handle a potential error and 2. Check if `result != ""`.

Reply:
Before removing for loop we had the same logic [here](https://github.com/wakatime/semver-action/blob/ea21bc32f18c09c5d3d2f88070b24f6837ec749f/pkg/git/git.go#L161).

Reply:
Alright, thanks for the additional info. Then I would suggest the following:
```suggestion
	if err != nil || result == "" {
             return c.Clean(c.Run("-C", c.repoDir, "describe", "--tags", "--abbrev=0"))
        }
        
        return result, nil
```
This will correctly handle the case, when git command returns an error. In this case we should ignore first return parameter and execute the fallback command.